### PR TITLE
Redemption features to old API

### PR DESCRIPTION
**Pull request by vantuz-subhuman**
Added couple of functions to `wasm-wallet/lib.rs` implementing required functionality:
1. Converting private redemption key to redemption address
2. Producing a redemption transaction

### CODE REVIEW

File: wallet-wasm/src/lib.rs [hunk 1]
```diff
@@ -1254,3 +1255,71 @@ pub extern "C" fn decrypt_with_password(
...
+    let address = address::ExtendedAddr::new(addr_type, addr_data, addr_attr));
+    let address_bytes = cbor!(address).unwrap();
+    unsafe { write_data(&address_bytes, out) }
+    return address_bytes.len() as u32;
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct WalletRedeemInput {
+    protocol_magic: cardano::config::ProtocolMagic,
+    redemption_key: redeem::PrivateKey,
```
Comment (vantuz-subhuman) [inline wallet-wasm/src/lib.rs]: Am I getting it right that caller might pass `protocol_magic` as integer number, and `redemption_key` as string or byte-array, and they will be deserialised into right types after calling `input_json!`?

File: wallet-wasm/src/lib.rs [hunk 2]
```diff
@@ -1254,3 +1255,71 @@ pub extern "C" fn decrypt_with_password(
...
+    let &mut txbuilder = txbuild::TxBuilder::new();
+    txbuilder.add_input(&data.input.convert(), data.output.value.0);
+    txbuilder.add_output_value(&data.output.convert());
+    let tx: tx::Tx = jrpc_try!(
+        output_ptr,
+        txbuilder.make_tx()
+            .map_err(|e| JsValue::from_str(&format! {"{:?}", e}))
+    );
+    let witness = tx::TxInWitness::redeem(
+        data.protocol_magic, &data.redemption_key, &tx.0.id());
```
Comment (vantuz-subhuman) [inline wallet-wasm/src/lib.rs]: Here I've imagined how new functionality for redemption signing might look:
image

But of course will update to actual new functionality when @NicolasDP adds `signRedeem` to new API

File: wallet-wasm/src/lib.rs [hunk 3]
```diff
@@ -1254,3 +1255,71 @@ pub extern "C" fn decrypt_with_password(
...
+    private_ptr: *const c_uchar,
+    protocol_magic: u32,
+    out: *mut c_uchar,
+) -> u32 {
+    let priv_key = unsafe {
+        let slice: &[u8] = std::slice::from_raw_parts(private_ptr, redeem::PRIVATEKEY_SIZE);
+        redeem::PrivateKey::from_slice(slice).unwrap()
+    };
+    let pub_key = priv_key.public();
+    let addr_type = address::AddrType::ATRedeem;
```
Comment (rooooooooob) [inline wallet-wasm/src/lib.rs]: What about redeem_pubkey_to_txid() in cardano's tx.rs?

Comment (vantuz-subhuman) [reply wallet-wasm/src/lib.rs]: Oh this is awesome 👍 Simplified the function

**Comment by vantuz-subhuman** [comment]
@NicolasDP, ready for final review and merge

**Pull request merged**